express_relay/sdk/python/express_relay_utils/express_relay_types.py

```diff
@@ -0,0 +1,318 @@
+    amount: IntString
+    executor: Address
+    permission_key: HexString
+    signature: SignedMessageString
+    valid_until: IntString
+
+    model_config = {
+        "arbitrary_types_allowed": True,
+    }
+
+
+class OpportunityParamsV1(BaseModel):
+    """
+    Attributes:
+        target_calldata: The calldata for the contract call.
+        chain_id: The chain ID to bid on.
+        target_contract: The contract address to call.
+        permission_key: The permission key to bid on.
+        buy_tokens: The tokens to receive in the opportunity.
+        sell_tokens: The tokens to spend in the opportunity.
+        target_call_value: The value to send with the contract call.
+        version: The version of the opportunity.
+    """
+
+    target_calldata: HexString
+    chain_id: str
+    target_contract: Address
+    permission_key: HexString
+    buy_tokens: list[TokenAmount]
+    sell_tokens: list[TokenAmount]
+    target_call_value: IntString
+    version: Literal["v1"]
+
+
+class OpportunityParams(BaseModel):
+    """
+    Attributes:
+        params: The parameters of the opportunity.
+    """
+
+    params: Union[OpportunityParamsV1] = Field(..., discriminator="version")
+
+
+class Opportunity(BaseModel):
+    """
+    Attributes:
+        target_calldata: The calldata for the contract call.
+        chain_id: The chain ID to bid on.
+        target_contract: The contract address to call.
+        permission_key: The permission key to bid on.
+        buy_tokens: The tokens to receive in the opportunity.
+        sell_tokens: The tokens to spend in the opportunity.
+        target_call_value: The value to send with the contract call.
+        version: The version of the opportunity.
+        creation_time: The creation time of the opportunity.
+        opportunity_id: The ID of the opportunity.
+    """
+
+    target_calldata: HexString
+    chain_id: str
+    target_contract: Address
+    permission_key: HexString
+    buy_tokens: list[TokenAmount]
+    sell_tokens: list[TokenAmount]
+    target_call_value: IntString
+    version: str
+    creation_time: IntString
+    opportunity_id: UUIDString
+
+    supported_versions: ClassVar[list[str]] = ["v1"]
+    unsupported_version_error_text: ClassVar[str] = (
+        "Cannot handle opportunity version: {%s}. Please upgrade your client."
+    )
+
+    @model_validator(mode="after")
+    def check_version(self):
+        if self.version not in self.supported_versions:
+            raise Exception("Unsupported version")
+        return self
+
+    @classmethod
+    def process_opportunity_dict(cls, opportunity_dict: dict):
+        """
+        Processes an opportunity dictionary and converts to a class object.
+
+        Args:
+            opportunity_dict: The opportunity dictionary to convert.
+
+        Returns:
+            The opportunity as a class object.
+        """
+        try:
+            return cls.model_validate(opportunity_dict)
+        except Exception as e:
```

Comment:
instead of catching a general exception we can throw something custom like UnsupportedOpportunity() and just catch that here.